### PR TITLE
feat(factor-analysis): add factor analysis package

### PR DESCRIPTION
**Pull request by MemeSlayer27**
## WHY:

This PR adds a new factor analysis package that provides essential statistical functionality for analyzing questionnaire responses, including polychoric correlation computation and factor analysis. 

Fixes #[Issue number] (Please add the actual issue number)

### What has been changed

- Added new `@openvaa/factor-analysis` package
- Implemented polychoric correlation computation
- Implemented factor analysis with varimax rotation
- Added comprehensive unit tests with >90% coverage
- Added detailed documentation including API usage examples
- Included numerical stability improvements for edge cases

## Check off each of the following tasks as they are completed

- [x] I have reviewed the changes myself in this PR. Please check the [self-review document](https://github.com/OpenVAA/voting-advice-application/blob/main/docs/contributing/self-review.md)
- [x] I have added or edited unit tests.
- [x] I have run the unit tests successfully.
- [ ] I have run the e2e tests successfully. *(N/A - This is a utility package without UI components)*
- [x] I have tested this change on my own device.
- [ ] I have tested this change on other devices (Using Browserstack is recommended). *(N/A - This is a non-UI package)*
- [ ] I have tested my changes using the [WAVE extension](https://wave.webaim.org/extension/) *(N/A - This is a non-UI package)*
- [ ] I have tested my changes using keyboard navigation and screen-reading *(N/A - This is a non-UI package)*
- [x] I have added documentation where necessary.
- [x] Is there an existing issue linked to this PR? *(SWP [#676](https://github.com/OpenVAA/voting-advice-application/issues/676))*
- [x] I have cleaned up the commit history and checked the commits follow [the guidelines](https://github.com/OpenVAA/voting-advice-application/blob/main/docs/contributing/CONTRIBUTING.md#commit-your-update)

### CODE REVIEW

File: packages/factor-analysis/src/factor-analysis.ts [hunk 1]
```diff
@@ -0,0 +1,494 @@
+import { FactorAnalysisOptions, FactorAnalysisResult } from './types';
+import { eigs } from 'mathjs';
+import {
+  standardNormalCDF,
+  standardNormalPDF
+} from './utils/normal-distribution';
+
+export class FactorAnalysis {
```
Comment (kaljarv) [inline packages/factor-analysis/src/factor-analysis.ts:8]: Are you thinking that the class might be extended by subclasses (in which case a static class might make life difficult) or other class methods be exposed? I'm just wondering if there's a need to use a class at all.

File: packages/factor-analysis/src/types/index.ts
Comment (kaljarv) [inline packages/factor-analysis/src/types/index.ts:1]: Could you move these types next to the relavant files, i.e. to `factorAnalysis.type.ts` etc.

File: packages/factor-analysis/src/factor-analysis.test.ts
Comment (kaljarv) [inline packages/factor-analysis/src/factor-analysis.test.ts:1]: Could you use camelCase when naming all ts files, e.g., factorAnalysis.test.ts

File: packages/factor-analysis/src/polychoric/matrix.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        const { correlation } = polychoricCorrelation(var1, var2, {
+          maxIterations: 100,
+          tolerance: 1e-6
+        });
+
+        // Fill both upper and lower triangle (symmetric matrix)
+        matrix[i][j] = correlation;
+        matrix[j][i] = correlation;
+      } catch (error) {
+        console.warn(
```
Comment (kaljarv) [inline packages/factor-analysis/src/polychoric/matrix.ts:60]: Are we expecting `polychoricCorrelation` to throw even with valid data or should we just throw here?

File: packages/factor-analysis/src/factor-analysis.test.ts [hunk 3]
```diff
@@ -0,0 +1,949 @@
+import { FactorAnalysis, polychoricCorrelation, analyzeFactors } from './index';
+import { describe, test, expect, afterAll } from 'vitest';
+
+describe('Factor Analysis and Polychoric Correlation', () => {
+  describe('Polychoric Correlation', () => {
```
Comment (kaljarv) [inline packages/factor-analysis/src/factor-analysis.test.ts:5]: Could you split the tests to files next to the test targets, e.g., this part to `polychoric/correlation.test.ts`.

File: packages/factor-analysis/src/factor-analysis.ts [hunk 4]
```diff
@@ -0,0 +1,494 @@
...
+   */
+  private static readonly DEFAULT_OPTIONS: Required<FactorAnalysisOptions> = {
+    maxIterations: 100,
+    tolerance: 1e-4,
+    rotateFactors: true,
+    minEigenvalue: 1e-10,
+    regularization: 1e-6
+  };
+
+  public static compute(
```
Comment (kaljarv) [inline packages/factor-analysis/src/factor-analysis.ts:28]: Please convert to named params. It's a bit of a slog, but could you also do the same for the private methods and those in the utilities as well to avoid bugs in the future. Those with just one required input plus an optional options object of a different type can be excluded.

File: packages/factor-analysis/src/api.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  communalities: number[]; // Communality values for each question
+  converged: boolean; // Whether the algorithm converged
+}
+
+/**
+ * Performs factor analysis on questionnaire responses
+ * @param input Response matrix and options
+ * @returns Factor analysis results with question-factor mappings
+ */
+export function analyzeFactors(
```
Comment (kaljarv) [inline packages/factor-analysis/src/api.ts:24]: As long as we're not able to process undefined values, could we quickly validate the input in this level by checking all the values are nonmissing numbers?

**Review by kaljarv — APPROVED** [review]
Thanks, this looks very good and well documented! I just added comments related to style and file organization, but it's good to go after those changes.

**Review by kaljarv — APPROVED** [review]
👏
